pkg/volume/volume_linux_test.go

```diff
@@ -0,0 +1,162 @@
+// +build linux
+
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package volume
+
+import (
+	"os"
+	"syscall"
+	"testing"
+
+	v1 "k8s.io/api/core/v1"
+	utiltesting "k8s.io/client-go/util/testing"
+)
+
+type localFakeMounter struct {
+	path       string
+	attributes Attributes
+}
+
+func (l *localFakeMounter) GetPath() string {
+	return l.path
+}
+
+func (l *localFakeMounter) GetAttributes() Attributes {
+	return l.attributes
+}
+
+func (l *localFakeMounter) CanMount() error {
+	return nil
+}
+
+func (l *localFakeMounter) SetUp(mounterArgs MounterArgs) error {
+	return nil
+}
+
+func (l *localFakeMounter) SetUpAt(dir string, mounterArgs MounterArgs) error {
+	return nil
+}
+
+func (l *localFakeMounter) GetMetrics() (*Metrics, error) {
+	return nil, nil
+}
+
+func TestSkipPermissionChange(t *testing.T) {
+	always := v1.AlwaysChangeVolumePermission
+	onrootMismatch := v1.OnRootMismatch
+	tests := []struct {
+		description         string
+		fsGroupChangePolicy *v1.PodFSGroupChangePolicy
+		gidOwnerMatch       bool
+		permissionMatch     bool
+		sgidMatch           bool
+		skipPermssion       bool
+	}{
+		{
+			description:   "skippermission=false, policy=nil",
```

Comment:
Also add gidmatch = true cases for policy = nil/always for completeness.

Reply:
fixed